src/components/NavBar.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useContext, useEffect } from 'react';
+import GlobalContext from '../state/GlobalContext';
+import Button from '@material-ui/core/Button';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import IconButton from '@material-ui/core/IconButton';
+import { Menu as MenuIcon, AccountCircle as AccountCircleIcon } from '@material-ui/icons';
+import InputBase from '@material-ui/core/InputBase';
+import Switch from '@material-ui/core/Switch';
+import styled from 'styled-components';
+
+const StyledInputBase = styled(InputBase)`
+  background-color: rgb(255, 255, 255, 0.15);
+  border-radius: 4px;
+  margin-right: 5px;
+
+  > input {
+    color: white;
+    padding-left: 5px;
+  }
+`;
+
+const NavBar = ({ className, handleSearchButton }) => {
+  const { dispatch, state } = useContext(GlobalContext);
+  const handleSearchField = (event) => {
+    dispatch({
+      type: 'CHANGE_SEARCH_TEXT',
+      payload: {
+        searchText: event.target.value,
+      }
+    });
+  }
+
+  useEffect(() => {
+    console.log("valor state", state);
+  });
+
+  return (
+    <>
```

Comment:
The React fragment `<>` can be removed here because you are rendering just one node on the dom.  https://reactjs.org/docs/fragments.html